### ISSUE TITLE
Add support for floodlight control on relevant cameras

### ISSUE THREAD

**Issue by binarydev** [ThomasLomas/ha-starlinghomehub#99]
### Checklist

- [X] I have filled out the template to the best of my ability.
- [X] This only contains 1 feature request (if you have multiple feature requests, open one feature request for each feature request).
- [X] This issue is not a duplicate feature request of [previous feature requests](https://github.com/ThomasLomas/ha-starlinghomehub/issues?q=is%3Aissue+label%3A%22enhancement%22).

### Is your feature request related to a problem? Please describe.

StarlingHub exposes floodlight control for the "Nest Cam with Floodlight (2nd generation)" device, but it's not available for control via this integration and no mention of it on the checklist of current+pending supported features.

### Describe the solution you'd like

Add switch control for the floodlight on these cameras. The field name is "floodlightOn" per the [latest API docs](https://sidewinder.starlinghome.io/sdc/?_gl=1*gact0e*_gcl_au*MTM5NzEyNzAyLjE3MzQzMjMzNjQ.*_ga*MTAzMDEwNjEyNC4xNzM0MzIzMzY1*_ga_RB2T845B29*MTczNDMyMzM2NS4xLjEuMTczNDMyMzg5Ni4zNS4xLjkxOTc3MTcx#camera:~:text=close-,floodlightOn,-Boolean)

### Describe alternatives you've considered

Use the direct homekit integration instead.

### Additional context

Happy to help with testing and debugging as a fellow SWE!

**Comment by ThomasLomas** [comment]
This was added in v1.0.8. Please report any issues in a new issue. Thanks!

**Comment by binarydev** [comment]
Unfortunately your update did not work. It turns out that the public API docs were out of date and the attribute you added was only part of the v1 API. I used the link on my own device to review its local copy of API docs for the v2 API, and I see that floodlights were broken out as part of a new light device category. I've attached a copy of my local API docs, which are noted as last updated on Dec 3rd, 2024, API Level 3.3, using firmware v2024.51.3.
[starling-api-local-reference.zip](https://github.com/user-attachments/files/18254736/starling-api-local-reference.zip)

I've also prepared a PR that adds support for the light device category, and I've confirmed that it works with Google Home's floodlight devices along with any 3rd party lights exposed by the Starling API. It supports turning the lights on and off, as well as setting the brightness, rgb color, and color temperature.
